playbooks/leapp_preupgrade_ansible.yml

```diff
@@ -182,12 +186,57 @@
           ansible.builtin.set_fact:
             error_count: "{{ report_content.entries | selectattr('groups', 'defined') | selectattr('groups', 'contains', 'error') | list | length }}"
 
+        - name: Init new empty entries for edited severity
+          ansible.builtin.set_fact:
+            report_content_edited: >-
+              {{
+                {
+                  "entries": []
+                }
+              }}
+
+        - name: Transform severities in leapp report to distinguish errors and inhibitors from other high risks entries
+          set_fact:
+            report_content_edited: >-
+              {{
+                report_content_edited | combine(
+                  {
+                    "entries": report_content_edited.entries | default([]) +
+                      [
+                        entry | combine(
+                          {"severity": "inhibitor"} if "error" in entry.groups | default([]) else
+                          {"severity": "inhibitor"} if "inhibitor" in entry.groups | default([]) else {}
+                        )
+                      ]
+                  }
+                )
+              }}
+          loop: "{{ report_content.entries | default([]) }}"
+          loop_control:
+            loop_var: entry
+
+        - name: Set message if inhibitors or errors present
+          set_fact:
+            message: >-
+              "The upgrade cannot proceed.
+              Your system has {{ error_count + inhibitor_count }} inhibitor{{ 's' if inhibitor_count + error_count != 1 else '' }}
+              out of {{ total_problems_count }} potential problems."
+          when: inhibitor_count != 0 or error_count != 0
+
+
+        - name: Set message if no inhibitors and no errors present
+          set_fact:
+            message: >-
+              {{ 'No problems found. The system is ready for upgrade.' if  total_problems_count == 0
+              else 'The upgrade can proceed. However, there is one or more warnings about issues that might occur after the upgrade.'}}
```

Comment:
Can we add this also to rhel7 pre/upgrade script ? To match the messages on both tasks.

Reply:
I tried to sync it in last commit, take a look please - I am confused about all the tasks myself, but it should be fine now. I kept the error+inhibitor = inhibitor approach for all of them

Reply:
Yes, looks good now. Thanks!